docs/components.rst

```diff
@@ -21,6 +21,93 @@ Each handler is associated to a service and is used to process different events
+It is specifically used to handle user workspaces and the user access to different other folders.
+The handler is responsible for creating/deleting a workspace if a user is created/deleted.
+
+Here are some typical folders found in the ``user_workspaces``:
+
+.. code-block:: bash
+
+    /user_workspaces/<user_name>/notebooks -> /jupyterhub_user_data/<user_name>
+    /user_workspaces/<user_name>/shapefile_datastore  # Managed by the `Geoserver` handler
+    /user_workspaces/public/wpsoutputs
+
+.. _filesystem_jupyterhub_user_data:
+
+JupyterHub user data
+####################
+
+A symlink is created in the user workspace to give access to the notebook folder of the user. The notebook folder is
+the folder that was originally mounted in `birdhouse` as the ``writable-workspace``. When adding the `Cowbird` component
+to `birdhouse`, the mounted ``writable-workspace`` becomes the user workspace managed by `Cowbird`
+(e.g.: ``/user_workspaces/<user_name>``) which requires a symlink to access the notebook folder originally used by
+`birdhouse` and `JupyterHub`.
+
+.. _filesystem_wps_outputs_data:
+
+Wps outputs data
+################
+
+The wps outputs data, generated by the different WPS birds and `Weaver`, is stored in a distinct data directory and must
+be made accessible to the user via his workspace. There are 2 types of data : public data, which must be accessible to
+every user, and user specific data, which must only be accessible to the related user.
+
+The public wps outputs data is made accessible by generating hardlinks from the wps outputs data directory to the
+user workspaces location (e.g.:``/user_workspaces/public/wpsoutputs``). When a `JupyterLab` instance is started via
+`birdhouse`, the folder containing the hardlinks will be mounted as volume, so that the users have access
+via their `JupyterLab` instance. The volume will be made read-only to prevent a user from modifying the public data.
+
+The user wps outputs data is made accessible by generating hardlinks from a wps outputs data directory containing user
+data to a subdirectory found in the related user's workspace. For example, with a source path
+``/wpsoutputs/<bird-name>/users/<user-id>/<job-id>/<output-file>``, a hardlink is generated at the path
+``/user_workspaces/<user-name>/wpsoutputs/<bird-name>/<job-id>/<output-file>``. The hardlink path uses a similar
+structure as found in the source path, but removes the redundant ``users`` and ``<user-id>`` path segments. The
+hardlink files will be automatically available to the user on a `JupyterLab` instance since the workspace is mounted as
+a volume. Any file that is found under a directory ``/wpsoutputs/<bird-name>/users/<user-id>/`` is considered to be
+user data and any outside file is considered public.
+
+The permissions found on the files are synchronized with the permissions found on `Magpie`. If `Magpie` uses a
+`secure-data-proxy` service, this service handles the permissions of those files. If a file does not have a
```

Comment:
Do you mean to add a mention that permissions should not be handled via the FileSystem and only from Magpie? That if a permission is changed directly on the FileSystem, it will be ignored?

Reply:
I will add a mention about this, because it's worth mentioning anyway, but let me know if you meant something else.